src/test/java/reposense/parser/ArgsParserTest.java

```diff
@@ -462,14 +462,13 @@ public void parse_repoAliases_sameResult() throws ParseException, HelpScreenExce
         Assert.assertEquals(repoAliasCliArguments, reposAliasCliArguments);
     }
 
-    @Test
-    public void parse_invalidRepoLocation_emptyRepoConfigurationList()
+    @Test (expected = InvalidLocationException.class)
```

Comment:
Even the logic is correct, this test should be in a different location though. This test is mean to test `ArgsParser`. Lets maintain this original test and add this test elsewhere.

Reply:
Thank you for your suggestion! I'll fix this by tomorrow.

Reply:
The original test was testing for the same component as the modified test. Basically they are both testing for the scenario where all repo locations supplied in the argument are invalid. The only change I made was the expected behaviour. The original test was verifying that the repo configuaration list is empty in method `RepoSense.getRepoConfigurations()` if all repo locations are invalid. I modified the method to make it throw exception when all locations are invalid. Therefore I also modified the test to expect an exception thrown. Since the test was in `ArgsParserTest.java` before I started to work on this component, I'm assuming that the development team decided to put it and treat it as something related to parsing user arguments. In this case, do I still need to move the test to somewhere else?

Reply:
I think @shuiyao-sg 's modification to the test is correct, but perhaps this test should not be in `ArgsParserTest`.

Majority of the tests in `ArgsParserTest` are testing whether the `CliArguments` object is correctly created by `ArgsParser`. However, this test seems to be checking the logic in RepoSense's driver class (i.e. `RepoSense.java`).

I think it should be moved into a separate file where we can test the logic in RepoSense's driver class.
However, perhaps this can be done in a subsequent PR, and this PR can proceed first?

I have created an issue on moving this tests [here](https://github.com/reposense/RepoSense/issues/1234)

Reply:
@eugenepeh, may we get your advice here?

I think that @shuiyao-sg 's modification to the test `parse_noValidRepoLocation_throwsInvalidLocationException` is appropriate.

However, I think this test should moved to a new test file, as I've explained above.
I think this PR can be merged first, and we can move the test in another PR?

Reply:
@fzdy1914 may we get your input on this discussion?

Reply:
@fzdy1914 may we get your thoughts when you are free? Thanks!

Reply:
Sorry for not replying, I think your suggestion is fine.